openfeature/provider/__init__.py

```diff
@@ -0,0 +1,61 @@
+import typing
+
+from openfeature.evaluation_context import EvaluationContext
+from openfeature.flag_evaluation import FlagResolutionDetails
+from openfeature.hook import Hook
+
+from .metadata import Metadata
+
+
+class FeatureProvider(typing.Protocol):
```

Comment:
```suggestion
class FeatureProvider(typing.Protocol):  # pragma: no cover
```
I think, this should disable coverage checking for the whole class, which we anyway don't need to do here

Reply:
Interesting suggestion! I usually use coverage tools to point me to what code has insufficient testing but never as a target itself. What is the policy in other OpenFeature projects?  Should we aim for 100% coverage here?

Reply:
We are already using `pragma: no cover` in some places, so it's not really a big deal

Reply:
> Interesting suggestion! I usually use coverage tools to point me to what code has insufficient testing but never as a target itself. What is the policy in other OpenFeature projects? Should we aim for 100% coverage here?

It's just a metric. We aim for high coverage, but coverage isn't always accurate. Some small PRs will have good testing but decrease overall coverage by 0.1% or something... As a maintainer you can bypass it. I recommend just using your judgement and not making it a blocker.